### ISSUE TITLE
Bug: [windows] searching for `"` turns into `^"break^"`

### ISSUE THREAD

**Issue by TheLeoP** [ibhagwan/fzf-lua#1193]
### RTFM Checklist

- [X] I have searched exisiting issues / discussions
- [X] I have read the Wiki including the Advanced section
- [ ] I have read `man fzf` / I am well versed in shell fzf

### Operating system

Windows 11 23H2

### Shell

powershell

### Neovim version (`nvim --version`)

NVIM v0.10.0-dev-3100+gd3fa88b70

### Fzf version (`fzf --version`)

0.52.0 (bcda25a)

### Output of `:lua print(os.getenv('FZF_DEFAULT_OPTS'))`

nil

### Is the problem reproducible with `mini.sh`?

- [ ] My issue is reproducible with `mini.sh`
- [ ] My issue **IS NOT** reproducible with `mini.sh`
- [X] I have not tested with `mini.sh` (not relevant, requires LSP, Windows, etc)

### Fzf-lua configuration

```lua
require('fzf-lua').setup({})
```

### Describe the bug / steps to reproduce

1. `FzfLua live_grep debug=true`
2. Type a single `"` into the prompt
3. `FzfLua` shows that the command being executed is `rg --column --line-number --no-heading --color=always --smart-case --max-columns=4096 -e ^"break^"`  (the important part being `^"break^"`)

Note: if a second `"` is typed, the escaping works as expected (i.e., rg searchs for `""` by receiving `-e ^"\^"\^"^"`)

**Comment by ibhagwan** [comment]
I noticed that too, this was introduced with the new windows escaping logic in fzf `0.51`, I haven’t has the chance to explore that enough to determine if this is a bug in fzf or fzf-lua, if you downgrade to 0.49 it will work as expected.

I will look deeper into this when I get a chance.

**Comment by ibhagwan** [comment]
@TheLeoP, this is an upstream issue, while I can remove functionality to make this work (do not display regex error messages / silent fail) I'd rather this be addressed properly by the upstream.

ref: https://github.com/junegunn/fzf/issues/3789

**Comment by ibhagwan** [comment]
@TheLeoP,  updated https://github.com/ibhagwan/fzf-lua/blob/main/README-Win.md, it is recommended to use fzf  `0.50.0` until the upstream is resolved.

In the meantime if you wish to remain with `0.52` https://github.com/ibhagwan/fzf-lua/commit/b64d2802d1349ae9c3d54062492856c2e071326a further adjusts for other changes in the new escaping added with `0.51/0.52` which helps the issue a bit, you can now search for `"` if you run:
```lua
:FzfLua live_grep exec_empty_query=true silent_fail=false stderr_to_stdout=false
```

Note that the above will still have oddities (described in https://github.com/junegunn/fzf/issues/3789#issuecomment-2107608321) due to the wrong escape sequence.

**Comment by ibhagwan** [comment]
@TheLeoP, the upstream was fixed with 0.52.1, I won’t be able to this before until tomorrow, if you happen to rest the latest commit of both fzf-lua and 0.52.1 let me know if this is solved or if there are other issues with special chars?

**Comment by ibhagwan** [comment]
@TheLeoP, everything should work after you update fzf to `0.52.1`.

Also recommended to update fzf-lua to https://github.com/ibhagwan/fzf-lua/commit/3131cb41c19877d661172bd09c8e3a1d38e171dd.

Lmk if otherwise?
